application-ldapuserimport-api/src/main/java/com/xwiki/ldapuserimport/internal/XWikiLDAPUtilsHelper.java

```diff
@@ -54,72 +53,132 @@ public final class XWikiLDAPUtilsHelper
      */
     public static final String LDAP_BASE_DN = "ldap_base_DN";
 
-    private static final String FILTER_CLOSING_MARK = "))";
+    private static final String OR_FILTER_STARTING_MARK = "(|";
+
+    private static final String AND_FILTER_STARTING_MARK = "(&";
+
+    private static final String FILTER_ENDING_MARK = ")";
 
     private static final String LDAP_GROUP_CLASSES_KEY = "ldap_group_classes";
 
+    private static final String LDAP_GROUP_SEARCH_FIELDS_KEY = "ldap_group_searchfields";
+
+    private static final String LDAP_USER_CLASSES_KEY = "ldap_user_classes";
+
     private static final String LDAP_GROUP_CLASSES = "group,groupOfNames,groupOfUniqueNames,dynamicGroup,"
         + "dynamicGroupAux,groupWiseDistributionList,posixGroup,apple-group";
 
+    private static final String LDAP_GROUP_SEARCH_FIELDS = CN;
+
+    private static final String LDAP_USER_CLASSES = "*";
```

Comment:
The default setting for user classes has been switched to "*" instead of inetOrgPerson as this user class might not be used in all LDAP databases, and in particular Active Directory